framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -304,6 +304,9 @@ void MediaPlayerImpl::startPlayer()
 		return;
 	}
 
+	auto source = mInputHandler.getInputDataSource();
```

Comment:
if (mCurState == PLAYER_STATE_PAUSED)

Reply:
Added it

Reply:
Is it necessary to check return value as below?
```c
if (mCurState == PLAYER_STATE_PAUSED) {
	auto source = mInputHandler.getInputDataSource();
	if (set_audio_stream_out(source->getChannels(), source->getSampleRate(),
							 source->getPcmFormat()) != AUDIO_MANAGER_SUCCESS) {
		meddbg("MediaPlayer startPlayer fail : set_audio_stream_out fail\n");
		notifyObserver(PLAYER_OBSERVER_COMMAND_START_ERROR, PLAYER_ERROR_INTERNAL_OPERATION_FAILED);
		return;
	}
}
```

Reply:
I'm not sure...
If there is only 1 player, and we did pause & play, set_audio_stream_out is not needed so it returns fail.

Reply:
I also tested it, only PlayerA start -> pause -> resume, it works fine.
As you said, `set_audio_stream_out` is not needed in such case, 
but it doesn't return failure, becase of below code:
```c
if (card_config->status == AUDIO_CARD_PAUSE) {
	medvdbg("reset previous preparing\n");
	reset_audio_stream_out();
}
```

Reply:
I got it. I'll apply your code.
Thanks for review